sdk/core/core-client/src/utils.ts

```diff
@@ -50,8 +103,19 @@ export function flattenResponse(
   responseSpec: OperationResponseMap | undefined
 ): unknown {
   const parsedHeaders = fullResponse.parsedHeaders;
+
+  /**
+   * If body is not asked for, we return the response headers only. If the response
+   * has a body anyway, that body must be ignored.
+   */
+  if (fullResponse.request.method === "HEAD") {
+    return parsedHeaders;
+  }
+
   const bodyMapper = responseSpec && responseSpec.bodyMapper;
+  const isNullable = bodyMapper?.nullable ?? false;
```

Comment:
nit
```suggestion
  const isNullable = Boolean(bodyMapper?.nullable);
```
reads better?

Reply:
Done.